Add Interlogix Ultrasync Alarm System Integration (#15588)
home-assistant/home-assistant.github.io: source/_integrations/ultrasync.markdown

---
title: Interlogix UltraSync Hub
description: Instructions on how to integrate your Interlogix UltraSync Hub within Home Assistant.
ha_category:
  - Alarm
ha_iot_class: Local Polling
ha_release: 0.119
ha_config_flow: true
ha_codeowners:
  - '@caronc'
ha_domain: ultrasync
---

> The `ultrasync` platform will allow you to monitor and control your Interlogix and/or ComNav NX-595E UltraSync Hub from within Home Assistant and setup automation.

Reviewer: ```suggestion
The Interlogix UltraSync Hub integration will allow you to monitor and control your Interlogix or ComNav NX-595E UltraSync Hub from within Home Assistant.
```

> ## Configuration
> 
> Go to the integrations page in your configuration and click on new integration -> UltraSync.

Reviewer: We have a nice little include for this, that handles this all 🤘 

```suggestion
{% include integrations/config_flow.md %}
```

> **Note**: You can only be logged into the ZeroWire/UltraSync hub with the same user once; a subsequent login with the same user logs out the other. Since Home Assistant (HA) actively polls and maintains a login session to this Hub, it can prevent you from being able to log into at the same time elsewhere (via it's website). It is strongly recommended that you create a second user account on your Hub dedicated for just HA.

Reviewer: ```suggestion
**Note**: You can only be logged into the ZeroWire/UltraSync hub with the same user once; a subsequent login with the same user logs out the other. Since Home Assistant (HA) actively polls and maintains a login session to this Hub, it can prevent you from being able to log in at the same time elsewhere (via its website). It is strongly recommended that you create a second user account on your Hub dedicated to just HA.
```

## Sensor

> This component will create these sensors in the format of `{ultrasync_hubname}_{sensor}`;  The below example assumes you accept the default name of `UltraSync` (which is still represented in lowercase):

Reviewer: It sounds like you are trying to describe the resulting entity IDs, however, those are not guaranteed.

Instead, you could just describe the type of sensors created (without using entity IDs).

- `ultrasync_area1state`: The Area 1 State
- `ultrasync_area2state`: The Area 2 State
- `ultrasync_area3state`: The Area 3 State
- `ultrasync_area4state`: The Area 4 State

> There are several states each sensor can be at, but usually they will be one of the following: `Unknown`, `Ready`, `Not Ready`, `Armed Stay`, and `Armed Away`.  The `Unknown` state is assigned to sensors that are not reporting; they usually sit in the spots of the Area's you're not monitoring.

Reviewer: ```suggestion
There are several states each sensor can be at, but usually, they will be one of the following: `Unknown`, `Ready`, `Not Ready`, `Armed Stay`, and `Armed Away`.  The `Unknown` state is assigned to sensors that are not reporting; they usually sit in the spots of the areas you're not monitoring.
```

Reviewer: Additionally, those are states as shown in the UI, they are not the actual state.

## Event Automation

> When an Zone or Sensor changes it's state an event usable for automation is triggered on the Home Assistant Bus.

Reviewer: I like this section, it helps users. However, have you considered adding support for device automations? That would make this all available via the UI (without needing the know the actual events and other technical details).

Reviewer: ```suggestion
When a Zone or Sensor changes its state an event usable for automation is triggered on the Home Assistant bus.
```

Possible events are:

- `ultrasync_sensor_update`: The event includes the sensor no, name, and new status it changed to.
- `ultrasync_area_update`: The event includes the area no, name, and new status it changed to (if it did). **Note**: Area's are sent periodic heartbeats in which case the state will not change at all.

> Example automation to send a message via [Apprise](https://www.home-assistant.io/integrations/apprise/) on a sensor change in your home:

Reviewer: ```suggestion
Example automation to send a message via [Apprise](/integrations/apprise/) on a sensor change in your home:
```

{% raw %}

```yaml
- alias: House Movement
  trigger:
    platform: event
    event_type: ultrasync_sensor_update
  action:
    service: notify.apprise
    data:
      title: "Sensor activated!"
      message: "{{trigger.event.data.name}} has new status {{trigger.event.data.status}}"
```

{% endraw %}

## Services

Available services:

- `stay`: Set alarm scene to Stay Mode
- `away`: Set alarm scene to Away Mode (fully activate Alarm)

> - `disarm`: Disarm the alarm

Reviewer: Shouldn't this be implementing the standard services of the `alarm_control_panel`?
